app/src/main/java/com/github/se/eduverse/ui/search/SearchProfileScreen.kt

```diff
@@ -0,0 +1,146 @@
+package com.github.se.eduverse.ui.search
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import com.github.se.eduverse.R
+import com.github.se.eduverse.model.Profile
+import com.github.se.eduverse.viewmodel.ProfileViewModel
+import com.github.se.eduverse.viewmodel.SearchProfileState
+
+const val TAG_SEARCH_FIELD = "search_field"
+const val TAG_LOADING_INDICATOR = "loading_indicator"
+const val TAG_NO_RESULTS = "no_results"
+const val TAG_IDLE_MESSAGE = "idle_message"
+const val TAG_ERROR_MESSAGE = "error_message"
+const val TAG_PROFILE_LIST = "profile_list"
+const val TAG_PROFILE_ITEM = "profile_item"
+const val TAG_PROFILE_IMAGE = "profile_image"
+const val TAG_PROFILE_USERNAME = "profile_username"
+const val TAG_PROFILE_STATS = "profile_stats"
+
+@Composable
+fun SearchProfileScreen(viewModel: ProfileViewModel, onProfileClick: (String) -> Unit) {
+  var searchQuery by remember { mutableStateOf("") }
+  val searchState by viewModel.searchState.collectAsState()
+
+  Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
+    TextField(
+        value = searchQuery,
+        onValueChange = {
+          searchQuery = it
+          viewModel.searchProfiles(it)
+        },
+        modifier = Modifier.fillMaxWidth().padding(bottom = 16.dp).testTag(TAG_SEARCH_FIELD),
+        placeholder = { Text("Search profiles...") },
+        leadingIcon = { Icon(Icons.Default.Search, contentDescription = null) },
+        singleLine = true)
+
```

Comment:
I thing you should add a trailingIcon with a cross (for example) which allow to close the search page. For example : 

```
trailingIcon = { 
  IconButton(onClick = { navigationAction.goBack() }) {
    Icon(Icons.Default.Close, "close")
  } 
},
```